lib/block_view.go

```diff
@@ -654,9 +659,28 @@ func (bav *UtxoView) CopyUtxoView() (*UtxoView, error) {
 
 	newView.TipHash = bav.TipHash.NewBlockHash()
 
+	// We pass the pointer through directly so everybody is updating the same cache.
+	newView.SnapshotCache = bav.SnapshotCache
```

Comment:
Is the cache thread safe, or does it needs its own lock? It's possible that one thread will read from it while another writes to it